File: fastq_2_fasta.py
import sys
filename = sys.argv[1]

with open(filename, "r") as ins:
    line_num = 0
    for line in ins:
        if (line_num % 4 == 0):
            print(">" + line[1:], end="")
        if (line_num % 4 == 1):
            print(line, end = "")
        line_num = line_num + 1

File: fastq_2_fasta.pl
use v5.10;
my $filename = $ARGV[0]; 

open(my $fh, '<', $filename)
  or die "Could not open file '$filename' $!";

my $line_num  = 0;
while (my $line= <$fh>) {
    if ($line_num % 4 == 0) {
        print(">". substr($line, 1));
        $line_num = 0;
    }
    if ($line_num % 4 == 1) {
        print($line);
    }

    $line_num++;
}
